server/index.js

```diff
@@ -38,6 +32,36 @@ app.get('/api/products', (req, res, next) => {
     .catch(err => next(err));
 });
 
+// View product details by productId
+app.get('/api/products/:productId', (req, res, next) => {
+  const { productId } = req.params;
+  if (!parseInt(productId, 10)) {
+    next(new ClientError('"productId" must be a positive integer', 400));
```

Comment:
Be sure to include a `return` statement to prevent your route from continuing to execute.